WPFHospitalEditor/SpecialistAppointmentSearchDialog.xaml.cs

```diff
@@ -0,0 +1,135 @@
+
+        private void Close_Click(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+
+        private void Search_Click(object sender, RoutedEventArgs e)
+        {
+            HospitalMap.clearAllResults();
+            if (InvalidInputForAppointment())
+            {
+                MessageBox.Show("Invalid input.");
+            }
+            else
+            {
+                String specialist = specialistComboBox.SelectedItem.ToString();
+                int specialistID = int.Parse(specialist.Split(" ")[0]);
+                Doctor chosenDoctor = doctorServerController.GetDoctorById(specialistID);
+                DateTime startDate = DateTime.ParseExact(dateFrom.Text + " 08:00", "MM/dd/yyyy HH:mm", null);
+                DateTime endDate = DateTime.ParseExact(dateTo.Text + " 16:00", "MM/dd/yyyy HH:mm", null);
+
+                TimeInterval timeInterval = new TimeInterval(startDate, endDate);
+                RecommendationPreference appointmentSearchPriority = setPriorityFromComboBox();
+
+                RecommendationRequestDto recommendationRequestDto = new RecommendationRequestDto()
+                {
+                    DoctorId = chosenDoctor.Id,
+                    SpecialtyId = chosenDoctor.DepartmentId,
+                    TimeInterval = timeInterval,
+                    Preference = appointmentSearchPriority
+                };
+
+                if (CheckEquipmentExistance())
+                {
+                    HospitalMap.appointmentSearchResult = schedulingServerController.GetAppointments(recommendationRequestDto);
+                    SearchResultDialog appointmentDialog = new SearchResultDialog(hospitalMap, SearchType.AppointmentSearch);
+                    appointmentDialog.ShowDialog();
+                }
+                else
+                {
+                    MessageBox.Show("There is no room with required equipment!");
+                }
+
+                
+            }
+        }
+
+        private bool InvalidInputForAppointment()
+        {
+            if (specialistComboBox.Text.Equals("") || dateFrom.Text.Equals("") || dateTo.Text.Equals("") || equipmentComboBox.Text.Equals("") || priorityComboBox.Text.Equals("")) return true;
+            return false;
+        }
+
+        private RecommendationPreference setPriorityFromComboBox()
+        {
+            if (priorityComboBox.SelectedIndex == 0) return RecommendationPreference.Doctor;
+            return RecommendationPreference.Time;
+        }
+
+        private bool CheckEquipmentExistance()
+        {
+            for (int i = 0; i < HospitalMap.appointmentSearchResult.Count; i++)
+            {
+                int roomId = HospitalMap.appointmentSearchResult[i].RoomId;
+                List<EquipmentDto> equipmentDtos = equipmentServerController.GetEquipmentByRoomId(roomId).ToList();
+                return CheckEquipmentInRoomExistance(equipmentDtos);
+            }
+            return true;
```

Comment:
If I understand, when this method is called, appointmentSearchResult will probably be empty list, right? It won't be empty only if you first search appointments and then specialist appointments. 
And second thing, if list is not empty, you will call CheckEquipmentInRoomExistance method only for first element in list, and return it's value.